src/CoreWCF.WebHttp/src/CoreWCF/Description/WebHttpServiceModelCompat.cs

```diff
@@ -133,6 +136,9 @@ where outputProp.CanWrite
                 // strings, numbers, and enums.
                 // Therefore a simple cast expression should convert the enums simply.
                 let value = Expression.Convert(Expression.Property(inputExpression, inputProp), outputProp.PropertyType)
+                // We need to copy the IsXXXSetExplicitly properties last
+                // as they can be overwritten by the other setters.
+                orderby IsXXXSetExplicitly.IsMatch(inputProp.Name)
                 select (MemberBinding)Expression.Bind(outputProp, value);
```

Comment:
I think a RegEx is overkill. There's a fixed set of properties and I think you can just use `inputProp.Name.Contains("SetExplicitly")` as there are no other uses of "SetExplicitly" than the properties causing problems.

Reply:
I tried making this change and it works. The problem I saw was that without making WebHttpServiceModelCompat public, the tests as written won't compile. There are two solutions.
####Test indirectly
If you set OpenApiContractInfo.ResponseFormat one way, and set the attribute a different way, then get the swagger file, you can check what content type it advertises.

####Include WebHttpServiceModelCompat in test project
I believe you should be able to explicitly add the WebHttpServiceModelCompat source file to the test project and then call it. I don't think it has any dependencies on other internal types.

Reply:
@mconnew I've switch to using an indirect test, by checking the response header instead.